apps/src/lib/node/ledger/shims/abcipp_shim.rs

```diff
@@ -14,20 +18,22 @@ use tower_abci::{BoxError, Request as Req, Response as Resp};
 use tower_abci_old::{BoxError, Request as Req, Response as Resp};
 
 use super::super::Shell;
-use super::abcipp_shim_types::shim::{request, Error, Request, Response};
+use super::abcipp_shim_types::shim::request::{FinalizeBlock, ProcessedTx};
+#[cfg(not(feature = "ABCI"))]
+use super::abcipp_shim_types::shim::response::TxResult;
+use super::abcipp_shim_types::shim::{Error, Request, Response};
 use crate::config;
-use crate::node::ledger::shims::abcipp_shim_types::shim::request::{
-    BeginBlock, ProcessedTx,
-};
+
 
 /// The shim wraps the shell, which implements ABCI++.
 /// The shim makes a crude translation between the ABCI interface currently used
 /// by tendermint and the shell's interface.
 #[derive(Debug)]
 pub struct AbcippShim {
```

Comment:
It looks like this shim will become more general, we will use it for both ABCI and ABCI++? We could name it just `Shim` now and remove the docstring

Reply:
We currently do use it for both, but eventually we won't support ABCI anymore. At that point we will delete this struct altogether. Right now though, we only use ABCI in our testnets.